libs/core/config/include/hpx/config/constexpr.hpp

```diff
@@ -26,10 +27,10 @@
 
 /// This macro evaluates to ``inline constexpr`` if the compiler supports it,
 /// ``constexpr`` otherwise.
-#ifdef HPX_HAVE_CXX17_INLINE_CONSTEXPR_VARIABLE
-#define HPX_INLINE_CONSTEXPR_VARIABLE inline constexpr
+#if defined(HPX_COMPUTE_DEVICE_CODE)
+#define HPX_INLINE_CONSTEXPR_VARIABLE HPX_DEVICE static const
 #else
-#define HPX_INLINE_CONSTEXPR_VARIABLE constexpr
+#define HPX_INLINE_CONSTEXPR_VARIABLE inline constexpr
```

Comment:
What I'm worried about with this change is that it makes *everything* that is an `HPX_INLINE_CONSTEXPR_VARIABLE` also a device variable, which we don't necessarily want. Perhaps we want a separate `HPX_HOST_DEVICE_INLINE_CONSTEXPR_VARIABLE` as an alternative to `HPX_HOST_DEVICE HPX_INLINE_CONSTEXPR_VARIABLE` so that one can opt-in to this where needed?